### ISSUE TITLE
🐛 Control flow is panicking

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#3778]
### Environment information

```block
Main
```

### What happened?

Our Ecosystem CI detected a panic on the [following file](https://github.com/wevm/wagmi/blob/main/packages/cli/src/commands/generate.ts). The panics arise from the Control Flow Graph.

I managed to reduce the bug to the following code:

```js
if (xs) {}
for (const y of ys) {
  () => {
    for (const z of zs) {
      break;
    }
  };
}
```

Here is the reported panic:

```shell
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: /home/conaclos/project/biome/crates/biome_control_flow/src/lib.rs:35:21
Thread Name: biome::worker_1
Message: index out of bounds: the len is 4 but the index is 5

index.js internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 4 but the index is 5
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
```

### Expected result

Biome should not panic

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Strangely, I can no longer reproduce the bug. Closing.
